rootfs/api/models/app.py

```diff
@@ -970,8 +970,13 @@ def whitelist(self, whitelist):
         service = self._fetch_service_config(self.id)
 
         try:
-            addresses = ",".join(address for address in whitelist)
-            service['metadata']['annotations']['router.deis.io/whitelist'] = addresses
+            if whitelist:
+                addresses = ",".join(address for address in whitelist)
+                service['metadata']['annotations']['router.deis.io/whitelist'] = addresses
+            elif 'router.deis.io/whitelist' in service['metadata']['annotations']:
+                service['metadata']['annotations'].pop('router.deis.io/whitelist', None)
```

Comment:
pop should fix https://github.com/deis/router/issues/246 as i don't want to have the value empty.